### ISSUE TITLE
Upgrade Natural Gallery JS to v10.0.1

### ISSUE THREAD

**Issue by PowerKiKi** [Ecodev/natural-gallery-typo3#38]
In the past 7 years, Natural Gallery JS [changed a lot](https://github.com/Ecodev/natural-gallery-js/compare/2.5.0...10.0.1).  It move to a pure ESM package. The overall size was reduced. New features are available. And overall it matured quite a bit.

It seems like a good time to upgrade Natural Gallery JS, while also upgrading this extension for the next TYPO3.

@fabarea, @Lebeau09 is it something you could include in work ?

@sambaptista, can you confirm it is a good idea ?

**Comment by sambaptista** [comment]
Obviously, the last version is way more modern and complete. A total refactor has been done since then. 

Appart from technical details, optimisations and probably unused new features in typo3, there is a new one that change drastically the user experience. 

By the time, the gallery was requiring the complete collection of images with dimensions. This was forcing typo to generate all images of a collection, freezing the page during many seconds and stressing the server. 

The new gallery autonomously decides what images it requires and asks the parent for the relevant images on infinite scroll mode. The required images are displayed or buffered  and lazy loaded relatively to their situation and user behavior.

**Comment by PowerKiKi** [comment]
@fabarea, @compaoreh338 quel est votre avis sur ce travail de mettre à jour la librairie JS ?
